### PR TITLE
Move some constants to limits.hpp

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: lib/fizzy/limits.hpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+namespace fizzy
+{
+namespace
+{
+constexpr unsigned page_size = 65536;
+// Set hard limit of 256MB of memory.
+constexpr unsigned memory_pages_limit = (256 * 1024 * 1024ULL) / page_size;
```
Comment (axic) [inline lib/fizzy/limits.hpp]: @gumb0 @chfast we could also camelcase these in this PR if you want.

Comment (gumb0) [reply lib/fizzy/limits.hpp]: I would like to, I think it's useful to see that a variable is a global const without jumping to definition

**Comment by axic** [comment]
@gumb0 @chfast how about this one? Keep it as it is, move to `limits.h`, fold into `types.h`?

**Comment by gumb0** [comment]
Keeping these in `limits.hpp` looks good.
Is unnamed namespace needed?

File: lib/fizzy/limits.hpp [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace fizzy
+{
+namespace
```
Comment (chfast) [inline lib/fizzy/limits.hpp]: Anonymous namespace should never be in a header.

Comment (axic) [reply lib/fizzy/limits.hpp]: That was a typo, not sure why it even compiled.

File: lib/fizzy/limits.hpp (hunk 1 above)
Comment (axic) [reply lib/fizzy/limits.hpp]: @gumb0 @chfast how about this one then?

File: lib/fizzy/limits.hpp (hunk 2 above)
Comment (chfast) [reply lib/fizzy/limits.hpp]: The effect on `constexpr` variables are neglectable - they have internal linkage anyway.

**Comment by axic** [comment]
On a second though `PageSize` probably belongs to `types.h`. Should we move both there?

**Comment by chfast** [comment]
> On a second though `PageSize` probably belongs to `types.h`. Should we move both there?

I don't know.

**Comment by axic** [comment]
Merging for now, just to get the camcelcase change in and make @gumb happy :wink:

**Comment by gumb0** [comment]
Thanks, I'll go happily rebase my stuff now.

**Comment by gumb0** [comment]
We should use it in `instantiate_test.cpp`, too

**Pull request merged**